cloud-optimized-geotiffs/cogs-details.qmd

```diff
@@ -1,28 +1,29 @@
 
-Scenario: If the COG is intended only for visualization, conversion to 3 band byte will improve performance. 
+Scenario: If the COG is intended only for visualization, conversion to 3 band byte will improve performance.
 
 > GDAL supported Data Types [list](https://gdal.org/drivers/raster/gtiff.html#gtiff-geotiff-file-format)
 
 
 ## Compression (aka File Size)
 
-The biggest benefit to compression is on the storage side. It’s always recommended to use a lossless compression method. **Deflate** or **LZW** are the most recommended compression algorithms, there are some choices that depend on the data type and distribution, and if the goal is maximum compression or not. Maximum compression does result in some performance loss.
+The biggest benefit to compression is on the storage side. It’s always recommended to use a lossless compression method unless the COG is intended to be used only for visualization. **Deflate** or **LZW** are the recommended compression algorithms for general use. **LERC** (Limited Error Raster Compression) provides very efficient data compression for float data if you're ok rounding to some precision you provide. (This could be useful, e.g. for elevation data where the source data is known to not have precision beyond a known value.) Some other compression algorithms may be preferred depending on the data type and distribution, and if the goal is maximum compression or not. Maximum compression does result in some performance loss.
 
 ## No Data
-Setting a no data value makes it clear to users and visualization tools what pixels are not actually data. For visualization this allows these pixels to be easily hidden (transparent). Historically many values have been used, 0, -9999, etc… The key is to make sure the GDAL flag for no data is set. It is also suggested that the smallest negative value be used instead of a random value. For byte and unsigned integers/floats this will be 0, if 0 has meaning in your data use a different value (like the max possible value). Having the right nodata flag set is important for overview generation.
+
+Setting a no data value makes it clear to users and visualization tools what pixels are not actually data. For visualization this allows these pixels to be easily hidden (transparent). Historically many values have been used, 0, -9999, etc… The key is to make sure the GDAL flag for no data is set. It is also suggested that the smallest negative value be used instead of a random value. For byte and unsigned integers data types this will be 0. For float data, setting NaN as the no data value is suggested. Make sure to use a no data value that does not have meaning in your data; otherwise use a different value (like the max possible value). Having the right nodata flag set is important for overview generation.
```

Comment:
> For float data, setting NaN as the no data value is suggested

👍 / 👎 @wildintellect @vincentsarago

Reply:
👍 NaN is ok... NaN are really in the data array :-)

Reply:
The only time I've seen an Issue is a NASA DAAC specifically asked us not do this when submitting to them. But I don't know a technical reason not to do it.

Reply:
you shouldn't use NaN if your data is not of type Float. That's the only issue I can think of